features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Overview.jsx

```diff
@@ -136,26 +141,66 @@ class Overview extends Component {
     handleTagChange(tags) {
         const api = new Api();
         const { apiUUID } = this;
+        const currentAPI = this.state.api;
         const promisedApi = api.get(apiUUID);
         promisedApi
-            .then((response) => {
-                const apiData = JSON.parse(response.data);
+            .then((getResponse) => {
+                const apiData = JSON.parse(getResponse.data);
                 apiData.tags = tags;
                 const promisedUpdate = api.update(apiData);
-                promisedUpdate.catch((errorResponse) => {
-                    console.error(errorResponse);
-                    Alert.error('Error occurred while updating tags');
-                });
+                promisedUpdate
+                    .then((updateResponse) => {
+                        this.setState({ api: updateResponse.obj });
+                    })
+                    .catch((errorResponse) => {
+                        console.error(errorResponse);
+                        Alert.error('Error occurred while updating tags');
+                        this.setState({ api: getResponse.obj });
+                    });
+            })
+            .catch((errorResponse) => {
+                console.error(errorResponse);
+                Alert.error('Error occurred while retrieving API');
+                this.setState({ api: currentAPI });
+            });
+    }
+
+    /**
+     * Handle transport update
+     *
+     * @param {Event} event Event
+     */
+    handleTransportChange(event) {
+        const api = new Api();
+        const { apiUUID } = this;
+        const currentAPI = this.state.api;
+        const promisedApi = api.get(apiUUID);
+        promisedApi
+            .then((getResponse) => {
+                const apiData = JSON.parse(getResponse.data);
```

Comment:
we could directly use `getResponse.obj` here ?

Reply:
Fixed